src/alohalytics.h

```diff
@@ -39,10 +39,11 @@ typedef std::map<std::string, std::string> TStringMap;
 
 class Stats final {
   std::string upload_url_;
-  // Stores already serialized and ready-to-append event with unique client id.
+  // Unique client id is inserted as a special event in the beginning of every archived file before gzipping it.
+  // In current implementation it is used to distinguish between different users in the events stream on the server.
   // NOTE: Statistics will not be uploaded if unique client id was not set.
-  std::string unique_client_id_event_;
-  MessagesQueue messages_queue_;
+  std::string unique_client_id_;
+  HundredKilobytesFileQueue messages_queue_;
```

Comment:
Current mobile implementation for iOS and Android will try to gzip and archive current file after hitting 100kb limit.

However, it can be a good idea to pass it as a constructor parameter and initialize from platform implementation.

> On Jul 2, 2015, at 20:22, Dima <notifications@github.com> wrote:
> 
> In src/alohalytics.h <https://github.com/KnowSheet/Alohalytics/pull/39#discussion_r33801197>:
> 
> >    // NOTE: Statistics will not be uploaded if unique client id was not set.
> > -  std::string unique_client_id_event_;
> > -  MessagesQueue messages_queue_;
> > +  std::string unique_client_id_;
> > +  HundredKilobytesFileQueue messages_queue_;
> Is Alohalytics being designed to be used exclusively with FileQueue-s of 100kB file size limit?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/39/files#r33801197>.
>